source/Halibut/Transport/Protocol/StreamTimeoutExtensionMethods.cs

```diff
@@ -26,8 +27,20 @@ public static async Task WithTimeout(this Stream stream, SendReceiveTimeout? tim
             }
             finally
             {
-                stream.ReadTimeout = currentReadTimeout;
-                stream.WriteTimeout = currentWriteTimeout;
+                try
+                {
+                    stream.ReadTimeout = currentReadTimeout;
```

Comment:
🤔 Just wondering, do other native .NET streams throw an exception when adjusting the Read/Write timeouts?

If not, then I wonder if an alternative would be to instead change the `NetworkTimeoutStream` to not throw an exception when changing `ReadTimeout` or `WriteTimeout`.

The current solution will be more safe, as it guarantees any read/write timeout alteration won't get in the way. But I couldn't help but wonder if we were guarding against something that shouldn't happen (and we get read/write timeout properties in `NetworkTimeoutStream` wrong instead).

Reply:
> If not, then I wonder if an alternative would be to instead change the NetworkTimeoutStream to not throw an exception when changing ReadTimeout or WriteTimeout.

It's up to the underlying stream and if it's a network stream, possibly the OS? I think they do throw if the stream is not usable, and you throw an unrelated, side effect error, hence why we wrapped it and rethrow the real error.

Reply:
If `NetworkTimeoutStream` doesn't throw, then do we risk getting into a situation where we swallow real errors and end up not correctly changing timeouts, but we think we have.